apps/lib/opt.c

```diff
@@ -590,6 +590,28 @@ int opt_long(const char *value, long *result)
     return 1;
 }
 
+/* Parse a long long, put it into *result; return 0 on failure, else 1. */
+int opt_long_long(const char *value, long long *result)
```

Comment:
As mentioned elsewhere (https://github.com/openssl/openssl/issues/22622#issuecomment-1793552376), a `size_t` would be my preferred choice.
```suggestion
/* Parse a size argument with an optional size suffix (e.g., 1024, 1k, 30M), put it into *result; return 0 on failure, else 1. */
int opt_size(const char *value, size_t *result)
```

Reply:
It's signed here, so `size_t` wouldn't be correct.  `int64_t` is the one to change to.

Reply:
I think if we allow for scaling (i.e. 256k, 800g, etc), opt_long is still perfectly acceptable no?